channeldb/interfaces.go

```diff
@@ -0,0 +1,198 @@
+package channeldb
+
+import (
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/batch"
+	"github.com/lightningnetwork/lnd/channeldb/models"
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+// GraphDB is an interface describing a persisted Lightning Network graph.
+//
+//nolint:interfacebloat
+type GraphDB interface {
```

Comment:
Perhaps this should instead live where it's used? Which IIUC in this PR woudl be in the new package created to handle graph management. This way, the package itself declares what it needs and it's up for callers to create a struct that's capable of meeting the interface.